src/Raven.Server/Rachis/Candidate.cs

```diff
@@ -75,18 +75,7 @@ private void Run()
                             continue; // we already voted for ourselves
                         var candidateAmbassador = new CandidateAmbassador(_engine, this, voter.Key, voter.Value);
                         _voters.Add(candidateAmbassador);
-                        try
-                        {
-                            _engine.AppendStateDisposable(this, candidateAmbassador);
-                        }
-                        catch (ConcurrencyException)
-                        {
-                            foreach (var ambassador in _voters)
-                            {
-                                ambassador.Dispose();
-                            }
-                            return; // we lost the election, because someone else changed our state to follower
-                        }
+                        _engine.AppendStateDisposable(this, candidateAmbassador); // concurrency exception here will dispose the current candidate and it ambassadors
```

Comment:
where is that disposed?

Reply:
When the state of the engine changes, IIRC